smartsim/settings/base.py

```diff
@@ -24,20 +24,22 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-from .._core.utils.helpers import (
-    expand_exe_path,
-    init_default,
-    is_valid_cmd,
-    fmt_dict
-)
+from .._core.utils.helpers import expand_exe_path, init_default, is_valid_cmd, fmt_dict
 from ..log import get_logger
 
 logger = get_logger(__name__)
 
 
 class RunSettings:
     def __init__(
-        self, exe, exe_args=None, run_command="", run_args=None, env_vars=None, **kwargs
+        self,
+        exe,
+        exe_args=None,
+        run_command="",
+        run_args=None,
+        env_vars=None,
+        reserved_run_args=None,
```

Comment:
Since the user is also expected to somtimes need to use the base `RunSettings` class, I don't love that this is expose to the user. I realize it's not the in the ``Experiment`` api, but maybe we could make this a class variable instead? For example, see the launchers for the ``supported_rs`` class variable. I think this might be cleaner since there aren't many reserved args it seems like.

Reply:
In retrospect making the `reserved_run_args` a class variable makes much more sense (why, for example, should different instances of `MpirunSettings` have different reserved arguments? Simple answer is they shouldn't!)

Change made in https://github.com/CrayLabs/SmartSim/pull/166/commits/a2d3a9414cced0fcc3a32b89229b54abf3fcbfdf